### ISSUE TITLE
Generate models with properties of built-in types from Cadl

### ISSUE THREAD

**Issue by annelo-msft** [Azure/autorest.csharp#2337]
### Models with properties of Cadl built-in types
This tracks the work to generate models (input/output/round-trip) from a Cadl API description that have properties of the following [Cadl built-in types](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#built-in-models):

#### Input Cadl file
[Cadl testserver models/primitive-properties](https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/models/primitive-properties)

#### Expected output
Files: 
- [x] [PrimitivePropertyModel.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.cs)
- [x] [PrimitivePropertyModel.Serialization.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.Serialization.cs)
- [x] Cadl types map to .NET types as described in type mapping below
- [x] `RequiredBodyDateTime` is serialized as `writer.WriteStringValue(RequiredBodyDateTime, "O")`
- [x] `RequiredDuration` is serialized as `writer.WriteStringValue(RequiredDuration, "P")`

### Basic type mappings
- [x] Cadl `string` -> .NET `string`
- [x] Cadl `int32` -> .NET `int`, 
    - note: resiliency work tracked separately by https://github.com/Azure/autorest.csharp/issues/2354#issuecomment-1191995113
- [x] Cadl `int64` -> .NET `long`
- [x] Cadl `safeint` -> .NET `long`, 
    - note: expect service-side validation that value is within int53 range
- [x] Cadl `float32` -> .NET `float`
    - note: there is no Azure API resiliency requirement for float to grow to a double
- [x] Cadl `float64` -> .NET `double`
- [x] Cadl `zonedDateTime` -> .NET `DateTimeOffset`
    - note: [RFC3339 if serialized in JSON body](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#json)
    - header serialization using [RFC7321 if serialized via a header](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#http-query-parameters-and-header-values), tracked by https://github.com/Azure/autorest.csharp/issues/2493
- [x] Cadl `duration` -> .NET `TimeSpan`
- [x] Cadl `boolean` -> .NET `boolean`

**Comment by annelo-msft** [comment]
Assigned to me during design specification phase; will reassign once design is complete for implmentation.

**Comment by annelo-msft** [comment]
Moving collection and dictionary properties to https://github.com/Azure/autorest.csharp/issues/2453

**Comment by annelo-msft** [comment]
Moving PlainDate and PlainTime to https://github.com/Azure/autorest.csharp/issues/2473, as they are out of scope for the August MVP.  @KrzysztofCwalina FYI

**Comment by archerzz** [comment]
See the test case here: https://github.com/Azure/autorest.csharp/blob/d969aba11793bab86d3ca39e7252e94171c5ed31/test/AutoRest.TestServerLowLevel.Tests/LowLevel/Generation/ModelWriterTests.cs#L30-L51

I think we can close this issue.

**Comment by annelo-msft** [comment]
@lirenhe / @archerzz, could you provide the output of the generator on https://github.com/Azure/cadl-ranch/tree/main/packages/cadl-ranch-specs/http/models/roundtrip-basic, so I can confirm that this issue is complete?

**Comment by annelo-msft** [comment]
@lirenhe / @archerzz, could you provide the output of the generator on https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/models/primitive-properties/main.cadl, so I can confirm that this issue is complete?
